roles/create-ai-environment/tasks/main.yaml

```diff
@@ -0,0 +1,289 @@
+
+      zip_file=`ls | grep zip`
+
+      unzip $zip_file
+      python '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/tools/analysis_yaml.py'
+    executable: /bin/bash
+
+- name: Get Runtime Info
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_runtime.txt'
+  register: runtime
+
+- name: Defind Parameters - runtime_name
+  shell: echo "{{ runtime.stdout | regex_findall('[java|go|python|c|c++]') | join("") }}"
+  register: runtime_name
+
+- name: Defind Parameters - runtime_version
+  shell: echo "{{ runtime.stdout | regex_replace('[java|go|python|c|c++]') }}"
+  register: runtime_version
+
+- name: Get Framework Name
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_name.txt'
+  register: framework
+
+- name: Get Framework Version
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/framework_version.txt'
+  register: framework_version
+
+- name: Get User Entry Point
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/entry_point.txt'
+  register: entrypoint
+
+- name: Check output folder variable exists
+  stat:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/output_folder.txt'
+  register: stat_ouput_folder_variable
+
+- name: Get output folder variable value
+  when: stat_ouput_folder_variable.stat.exists
+  shell: cat '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/output_folder.txt'
+  register: output_folder_name
+
+- name: Check output folder exists
+  when: output_folder_name is defined
+  stat:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/{{ output_folder_name.stdout }}'
+  register: stat_ouput_folder
+
+- name: Create output folder if non-exist
+  when: output_folder_name is defined and not stat_ouput_folder.stat.exists
+  file:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/{{ output_folder_name.stdout }}'
+    state: directory
+
+- name: Set training output folder name by file
+  when: output_folder_name is defined
+  set_fact:
+    training_output_folder_name: "{{ output_folder_name.stdout }}"
+
+- name: Generate new output folder if non-exist
+  when: output_folder_name is not defined
+  file:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/training_result'
+    state: directory
+
+- name: Set training output folder name by generation
+  when: output_folder_name is not defined
+  set_fact:
+    training_output_folder_name: "training_result"
+
+- name: Install {{ runtime.stdout }} Runtime
+  when: runtime_name
+  shell:
+    cmd: |
+      set -x
+      set -e
+      runtime_name={{ runtime_name.stdout }}
+      runtime_version={{ runtime_version.stdout }}
+      add-apt-repository -y ppa:deadsnakes/ppa
+      apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F60F4B3D7FA2AF80
+      apt update
+      if [ $runtime_name == 'python' ];then
```

Comment:
How about to use `conda` to manage all of these runtime and packages in each runtime?
https://docs.conda.io/en/latest/miniconda.html

Reply:
I think if we use conda seems a little heavy.. Also the job is only support 1 kind runtime version, not need the venv provided by conda for now, that means once a job arise, we will only install python/java... And the configured apt repo is specific for containing all necessary packages which we use.  So for this reason, I don't think it is different with apt we used now. But in the future, by increasing requirements, maybe we can introduce conda for fiting the new cases.

Reply:
@bzhaoopenstack and @wangxiyuan , that is just a suggestion about using `conda`, please think about it, because we will support more frameworks, runtimes and libraries in future, if...else block may be larger, some tools, like: conda will help to decrease some complexity about package management.